### PR TITLE
cleanup and partitioning and functions

### PR DESCRIPTION
**Pull request by Simson475**

### CODE REVIEW

**Review by Zzeeker — APPROVED** [review]

File: VirtualAssistantBusinessLogic/VirtualAssistantBusinessLogic/SparQL/XMLResponseDecoder.cs [hunk 1]
```diff
@@ -52,30 +43,72 @@ public class XMLResponseDecoder : IResponseDecoder
...
+                        AddValueToKey(results, xmlReader, id, sanitizedKey);
                     }
                 }
             }
             return results;
         }
+        /// <summary>
+        /// Finds the key from the XML parsing and checks whether the key already exists.
+        /// If it does not exist, it is added to results array
+        /// Either way we return the found key
```
Comment (BjoerklundJacob) [inline VirtualAssistantBusinessLogic/VirtualAssistantBusinessLogic/SparQL/XMLResponseDecoder.cs:56]: 1 - It does not find the key, it is given as an argument.
2 - It is not added to any array, it is a dictionary
3 - We already found the key. We return a sanitised version of the key.

The method name is also a bit weird. Maybe SanitizeAndAddKey() instead

**Review by BjoerklundJacob — CHANGES_REQUESTED** [review]
Har kommenteret hvad der er galt

File: VirtualAssistantBusinessLogic/VirtualAssistantBusinessLogic/SparQL/XMLResponseDecoder.cs [hunk 2]
```diff
@@ -52,30 +43,76 @@ public class XMLResponseDecoder : IResponseDecoder
...
-                        }
+                        string sanitizedKey = SanitizeAndAddKey(results, id, key);
+                        AddValueToKey(results, xmlReader, id, sanitizedKey);
                     }
                 }
             }
             return results;
         }
+        /// <summary>
+        /// sanitizes the key from the XML parsing and checks whether the key already exists.
```
Comment (BjoerklundJacob) [inline VirtualAssistantBusinessLogic/VirtualAssistantBusinessLogic/SparQL/XMLResponseDecoder.cs:54]:
```suggestion
        /// Sanitizes the key from the XML parsing and checks whether the key already exists.
```

File: VirtualAssistantBusinessLogic/VirtualAssistantBusinessLogic/SparQL/XMLResponseDecoder.cs [hunk 3]
```diff
@@ -52,30 +43,76 @@ public class XMLResponseDecoder : IResponseDecoder
...
+        /// If it does not exist, it is added to results
+        /// Either way we return the sanitized key
+        /// </summary>
+        /// <param name="results">the results dict to add keys to</param>
+        /// <param name="id"> id found from previous step</param>
+        /// <param name="key"> unsanitized key with label</param>
```
Comment (BjoerklundJacob) [inline VirtualAssistantBusinessLogic/VirtualAssistantBusinessLogic/SparQL/XMLResponseDecoder.cs:60]:
```suggestion
        /// <param name="results">the results dict to add keys to</param>
        /// <param name="id">id found from previous step</param>
        /// <param name="key">unsanitized key with label</param>
```

**Review by BjoerklundJacob — APPROVED** [review]

**Review by Zzeeker — APPROVED** [review]

**Pull request merged**